File: src/buttons.py
from machine import Pin

class Buttons:
    def __init__(self,config):
        self.pins = [Pin(pn,Pin.IN,Pin.PULL_UP) for pn in config['pins']]
    
    def read(self):
        return [p.value() == 0 for p in self.pins]

File: js/buttons.js
/* Button module */

function buttons(msgr, btns)
{
  this.msgr=msgr
  this.btns=btns||[]
  for (p in this.btns) {
    pinMode(p, 'input_pullup')
    setWatch(pressed.bind(this, p), p, {repeat:true, edge:'falling', debounce:25})
  }
}

function pressed(pin)
{
  let str = []
  for (p in this.btns) {
    if (p == pin || !digitalRead(p)) {
      str.push(this.btns[p])
    }
  }
  this.msgr.send("button/"+str.join("_"))
}

exports.init = function(msgr, btns) {
  return new buttons(msgr, btns)
}
